### ISSUE TITLE
Generate pipelines using a template engine

### ISSUE THREAD

**Issue by tegataiprime** [concourse/concourse#3844]
## What challenge are you facing?

We've built pipeline templates that adhere to our firm's security, risk management & legal compliance rules.

We have created several "approved" pipeline templates for different types of software component that our teams produce; e.g. Java/Spring Library, Java/Spring Boot Application, Docker Image, etc.

There are common jobs & tasks shared by these pipelines, and we'd like to be able to break up jobs & tasks into individual yaml files that can be assembled into a complete pipeline prior to publishing, to maximize re-use and to minimize duplication.

## What would make this better?

We could implement our own template driven pipeline assembler, using a templating engine, such as GoTPL (as used by helm) to introduce features such as scoped variables, functions and include files.  But we were thinking this would be a powerful feature if it were baked into Concourse.

## Are you interested in implementing this yourself?

Well....yeah, we could take a crack at it.  We would most likely create a standalone utility that would generate pipelines ready for validation by the fly cli.

What guidance would you folks give for extending the fly cli to include this behavior?

**Comment by vito** [comment]
This comes up a lot, but I really don't think we should implement our own templating solution or even integrate with a popular one. You may be interested in #1994, specifically https://github.com/concourse/concourse/issues/1994#issuecomment-472145314 which also includes some of the reasoning for my stance on templating. We haven't scheduled the work for #1994 yet so I'd be super happy to review a PR for it if you're interested, since it sounds in line with what you want to do, aside from the templating aspect itself. :slightly_smiling_face: 

The tl;dr is that I prefer a UNIXy approach, and would rather make sure `fly` is composable enough that the user can use whatever templating system they want. Maintaining a templating system can quickly become a burden for the project, and I'd rather us stay focused on solving CI problems rather than solving text/YAML generation problems (it seems like there are enough projects trying to do that).

Hope this helps!
